how-to/workspace-platform-starter/client/src/framework/platform/platform-splash.ts

```diff
@@ -0,0 +1,138 @@
+import type OpenFin from "@openfin/core";
+import { getManifestCustomSettings } from "../settings";
+import { isStringValue } from "../utils";
+
+let win: OpenFin.Window | undefined;
+
+/**
+ * Open a splash screen.
+ */
+export async function open(): Promise<void> {
+	const app = await fin.Application.getCurrent();
+	const manifest = await app.getManifest();
+
+	const customSettings = await getManifestCustomSettings();
+
+	const disabled = customSettings?.splashScreenProvider?.disabled ?? false;
+
+	if (!disabled) {
+		win = await fin.Window.create({
+			name: "platform-splash",
+			uuid: "platform-splash",
+			url: `${window.location.href.replace("provider.html", "splash.html")}`,
+			alwaysOnTop: true,
+			maximizable: false,
+			minimizable: false,
+			frame: false,
+			autoShow: false,
+			defaultCentered: true,
+			defaultWidth: customSettings?.splashScreenProvider?.width ?? 400,
+			defaultHeight: customSettings?.splashScreenProvider?.height ?? 130,
+			includeInSnapshots: false,
+			resizable: false,
+			saveWindowState: false,
+			showTaskbarIcon: false
+		});
+
+		if (win) {
+			const webWin = win.getWebWindow();
+			if (webWin) {
+				const title = customSettings.splashScreenProvider?.title ?? manifest.shortcut?.name ?? "OpenFin";
+				const icon =
+					customSettings.splashScreenProvider?.icon ??
+					manifest.platform?.icon ??
+					"../common/images/icon-blue.png";
+				let backgroundColor: string | undefined = customSettings.splashScreenProvider?.backgroundColor;
+				let textColor: string | undefined = customSettings.splashScreenProvider?.textColor;
+				let borderColor: string | undefined = customSettings.splashScreenProvider?.borderColor;
+
+				webWin.document.title = title;
+
+				const iconElem = webWin.document.querySelector<HTMLImageElement>("#icon");
+				if (iconElem) {
+					iconElem.src = icon;
+				}
+
+				const headingElem = webWin.document.querySelector<HTMLElement>("#heading");
```

Comment:
So this looks like someone could create their own splash page as long as they followed the element contract. Can you specify a custom loading page?

Reply:
In theory you could switch out the whole HTML page with your own content